zanata-war/src/main/java/org/zanata/action/MergeTransAction.java

```diff
@@ -0,0 +1,230 @@
+package org.zanata.action;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+import javax.faces.application.FacesMessage;
+import javax.faces.component.UIOutput;
+import javax.faces.event.AjaxBehaviorEvent;
+import javax.faces.event.ValueChangeEvent;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import org.apache.commons.lang.StringUtils;
+import org.jboss.seam.ScopeType;
+import org.jboss.seam.annotations.Create;
+import org.jboss.seam.annotations.In;
+import org.jboss.seam.annotations.Name;
+import org.jboss.seam.annotations.Scope;
+import org.jboss.seam.faces.FacesMessages;
+import org.jboss.seam.international.StatusMessage;
+import org.zanata.async.handle.MergeTranslationsTaskHandle;
+import org.zanata.common.EntityStatus;
+import org.zanata.dao.ProjectDAO;
+import org.zanata.dao.ProjectIterationDAO;
+import org.zanata.i18n.Messages;
+import org.zanata.model.HProject;
+import org.zanata.model.HProjectIteration;
+import org.zanata.ui.CopyAction;
+
+/**
+ * Handles action from merge_trans_modal.xhtml
+ *
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Name("mergeTransAction")
+@Scope(ScopeType.PAGE)
+public class MergeTransAction implements Serializable, CopyAction {
+
+    @Getter
+    private String targetProjectSlug;
+
+    @Getter
+    @Setter
+    private String targetVersionSlug;
+
+    @Getter
+    private String sourceProjectSlug;
+
+    @Getter
+    @Setter
+    private String sourceVersionSlug;
+
+    @Getter
+    @Setter
+    private boolean keepExistingTranslation;
+
+    @In
+    private ProjectDAO projectDAO;
+
+    @In
+    private ProjectIterationDAO projectIterationDAO;
+
+    @In
+    private MergeTranslationsManager mergeTranslationsManager;
+
+    @In
+    private CopyTransManager copyTransManager;
+
+    @In
+    private CopyVersionManager copyVersionManager;
+
+    @In
+    private Messages msgs;
+
+    private HProjectIteration targetVersion;
+
+    private HProject sourceProject;
+
+    public void setTargetProjectSlug(String targetProjectSlug) {
+        this.targetProjectSlug = targetProjectSlug;
+
+        /**
+         * This should only true during first instantiation. See pages.xml
+         * for setter of mergeTransAction.targetProjectSlug
+         */
+        if(sourceProjectSlug == null) {
+            setSourceProjectSlug(targetProjectSlug);
+        }
```

Comment:
Is it because we want the default source project to be the target project? We should state that explicitly so that future developers know why this code is here.